apps/backend/src/datasources/postgres/ReviewDataSource.ts

```diff
@@ -284,4 +284,50 @@ export default class PostgresReviewDataSource implements ReviewDataSource {
         return reviews.map((review) => createReviewObject(review));
       });
   }
+
+  async getAllUsersReviews(
+    fapIds: number[],
+    userId?: number,
+    callId?: number,
+    instrumentId?: number,
+    status?: ReviewStatus
+  ): Promise<Review[]> {
+    const fapReviewsFiltered = database
+      .select('fap_reviews.*')
+      .from('fap_reviews')
+      .modify((qb) => {
+        // sometimes the ID 0 is sent as a equivalent of all
+        if (callId) {
+          qb.join('proposals', {
+            'proposals.proposal_pk': 'fap_reviews.proposal_pk',
+          });
+          qb.where('proposals.call_id', callId);
+        }
+
+        // sometimes the ID 0 is sent as a equivalent of all
+        if (instrumentId) {
+          qb.join('instrument_has_proposals', {
+            'instrument_has_proposals.proposal_pk': 'fap_reviews.proposal_pk',
+          });
+          qb.where('instrument_has_proposals.instrument_id', instrumentId);
+        }
+
+        if (status !== undefined && status !== null) {
+          qb.where('fap_reviews.status', status);
+        }
+      })
+      .whereIn('fap_id', fapIds);
+
+    const query = `SELECT DISTINCT ON (proposal_pk) *  

```

Comment:
Fap proposals are reviewed by multiple reviewers, lets say proposal_1 has two reviewers reviewer_1 and reviewer_2.
The query ensures that when reviewer_1 logs into the proposal system, it returns only reviewer_1 as reviewer against proposal_1. 
Similarly when reviewer_2 logs into the system, the query returns reviewer_2 against proposal_1.

Reply:
Ok I think I understand how the query is working, but I think it is quite confusing to work out the logic. Would you be able to put in a comment to explain how this query is working in case future people are as confused as I was.

Reply:
@TCMeldrum and @Scott-James-Hurley please review